### PR TITLE
Credentials dropdown for HTTP Request node

### PR DESCRIPTION
**Pull request by ivov**
From https://github.com/n8n-io/n8n/pull/3209, changed base to reversioned HTTPRN

### CODE REVIEW

File: packages/editor-ui/src/components/ParameterInputList.vue [hunk 1]
```diff
@@ -258,7 +265,21 @@ export default mixins(
 </script>
 
 <style lang="scss">
-.paramter-input-list-wrapper {
+.parameter-input-list-wrapper {
+	display: flex;
+	flex-direction: column;
+	padding-top: var(--spacing-xs);
+
+	&.has-reordered-credentials-input {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/ParameterInputList.vue:273]: not a fan of this approach. we should think of a less-hacky global way to do this... would rather leave as is rather than this. 
like trying to find position automatically if possible or even adding a property flag `showCredentialsAfter: true`

File: packages/editor-ui/src/components/NodeCredentials.vue [hunk 2]
```diff
@@ -93,6 +98,13 @@ export default mixins(
 		...mapGetters('credentials', {
 			credentialOptions: 'allCredentialsByType',
 		}),
+		isProxyCredentialType(): boolean {
+			const proxyCredentialNodeTypes = ["n8n-nodes-base.httpRequest"];
```
Comment (mutdmour) [inline packages/editor-ui/src/components/NodeCredentials.vue:102]: would need node version. is there possible to have more than one proxy type? should this live in node json instead?

File: packages/editor-ui/src/components/NodeCredentials.vue [hunk 3]
```diff
@@ -106,6 +118,10 @@ export default mixins(
 		credentialTypesNodeDescription (): INodeCredentialDescription[] {
 			const node = this.node as INodeUi;
 
+			if (this.isProxyCredentialType) {
+				return [this.$store.getters['credentials/getCredentialTypeByName'](this.node.parameters.nodeCredentialType)];
+			}
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/NodeCredentials.vue]: <details>
  <summary>View screencaps</summary>

Executions:
image

Credential deleted:
image
</details>

File: packages/editor-ui/src/components/NodeCredentials.vue (hunk 2 above)
Comment (ivov) [reply packages/editor-ui/src/components/NodeCredentials.vue]: HTTP Request node is only one that may access credentials not defined on itself. A simple check for node type should be enough for now.

File: packages/editor-ui/src/components/ParameterInputList.vue (hunk 1 above)
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputList.vue:269]: Refactored as `showCredentialsAfter` prop but open to suggestions if it's worth finding position automatically.

Comment (mutdmour) [reply packages/editor-ui/src/components/ParameterInputList.vue:269]: let's attempt it... just time box it

File: packages/editor-ui/src/components/NodeCredentials.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/components/NodeCredentials.vue]: should not this show error state if credential has been deleted?

[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/NodeCredentials.vue]: Might be related to pending issue mentioned in ticket: `Generic and typed creds not marked as required at subsequent selections`

[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/NodeCredentials.vue]: Solved

File: packages/editor-ui/src/components/ParameterInputList.vue (hunk 1 above)
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputList.vue]: See [75eea89](https://github.com/n8n-io/n8n/pull/3222/commits/75eea89273b854110fa6d1f96c7c1d78dd3b0731)

File: packages/editor-ui/src/components/mixins/nodeHelpers.ts [hunk 4]
```diff
@@ -394,3 +439,41 @@ export const nodeHelpers = mixins(
...
+ */
+function selectedCredsAreUnusable(node: INodeUi, credentialType: string) {
+	return node.credentials === undefined || Object.keys(node.credentials).includes(credentialType) === false;
+}
+
+/**
+ * Whether the node's selected credentials of the specified type
+ * can no longer be found in the database.
+ */
+function selectedCredsWereDeleted(
```
Comment (mutdmour) [inline packages/editor-ui/src/components/mixins/nodeHelpers.ts:459]: `credentialsDontExist` is more accurate

File: packages/editor-ui/src/components/mixins/workflowHelpers.ts [hunk 5]
```diff
@@ -326,10 +327,17 @@ export const workflowHelpers = mixins(
...
+					const fullAccess = node.type === HTTP_REQUEST_NODE_TYPE && node.typeVersion === 2;
+
 					// Add the node credentials if there are some set and if they should be displayed
 					if (node.credentials !== undefined && nodeType.credentials !== undefined) {
 						const saveCredenetials: INodeCredentials = {};
 						for (const nodeCredentialTypeName of Object.keys(node.credentials)) {
+							if (fullAccess) {
+								saveCredenetials[nodeCredentialTypeName] = node.credentials[nodeCredentialTypeName];
+								continue;
+							}
```
Comment (mutdmour) [inline packages/editor-ui/src/components/mixins/workflowHelpers.ts:339]: why is this needed?

Comment (ivov) [reply packages/editor-ui/src/components/mixins/workflowHelpers.ts]: To prevent cred ref from being lost. `find()` on the next line will never find anything for HTTPRN because this is the only node able to access cred types not defined on itself.

https://linear.app/n8n/issue/N8N-3354#comment-0ce9d056

**Review by mutdmour — APPROVED** [review]
